pootle/apps/virtualfolder/models.py

```diff
@@ -347,19 +275,156 @@ def get_adjusted_pootle_path(self, pootle_path):
         trail = pootle_path.replace(lead, '').lstrip('/')
         return '/'.join([lead, self.name, trail])
 
-    def get_translate_url(self, pootle_path, **kwargs):
-        """Get the translate URL for this virtual folder in the given path."""
-        adjusted_path = self.get_adjusted_pootle_path(pootle_path)
-        lang, proj, dp, fn = split_pootle_path(adjusted_path)
 
+class VFolderCachedTreeItemManager(models.Manager):
+    use_for_related_fields = True
+
+    def get_queryset(self):
+        return super(VFolderCachedTreeItemManager, self) \
+            .get_queryset().select_related('vfolder')
+
+
+class VFolderCachedTreeItem(models.Model, CachedTreeItem):
```

Comment:
@unho, let's don't speak about merging together `VirtualFolder` and `VirtualFolderCachedTreeItem`. 
Actually we have one `RootVirtualFolder` and its children.
What do you think if we replace `VirtualFolder` with `RootVirtualFolder(models.Model, CachedTreeItem)` and then `VirtualFolderCachedTreeItem` can become `VirtualFolder(models.Model, CachedTreeItem)`? 
And then even if we need to override `CachedTreeItem` that class can be called `VirtualFolderCachedTreeItem`.